Splitting RobotsEditor (#127557311)

TanVD opened the pull request:
Divides up robots-editor to auth, dashboard and editor service.
DB services should be started with exec:java command, new services based on tomcat servlet container and should be started with tomcat7:run-war command.

TanVD committed:
Splitting Robots Editor
1.Authentication added
2.Dashboard service added (fixes needed)

TanVD committed:
Splitting RobotsEditor
1.CSRF problem fixed
2.Pom fixed in DashboardService

TanVD committed:
Splitting RobotsEditor
1.EditorService created
2.Main functionality fixed
3.Property files for jsp and js created (contains addresses of services)
4.Minor improvements
TODO:
1.Delete robots-editor
2.Create service  presenting all other services through port 8080 (may have CORS problem)

TanVD committed:
Splitting RobotsEditor
1.RobotsEditor deleted
2.AuthService refactored
3.Minor improvements

TanVD committed:
Splitting RobotsEditor
1.Few styleguide fixes
2.Checkstyle fixes
3.Codacy fixes

TanVD committed:
Splitting RobotsEditor
1.Fixes for Travis

TanVD committed:
Splitting RobotsEditor
1.3d party lib installation added

TanVD committed:
Splitting RobotsEditor
1.Fixes for codacy
2.Fixes for Travis

TanVD committed:
Splitting RobotsEditor
1.RelativePath tag added

TanVD committed:
Splitting RobotsEditor
1.Fixed ln -s command for Travis

TanVD committed:
Splitting RobotsEditor
1.Fixed callTomcat test

TanVD committed:
Splitting RobotsEditor
1.In buildApp script parent install added

TanVD committed:
Trying to fix Travis build.

TanVD committed:
Merge branch 'master' into split-robotseditor

TanVD committed:
Log problem fixed. (Exclusions added in pom files, conflict resolved)

TanVD committed:
Merge branch 'master' into split-robotseditor

TanVD committed:
Trying to fix Travis problems: tomcat7-maven-plugin dependency now is explicit.

TanVD committed:
Added name of service to log messages.

TanVD committed:
Returning run-war instead run-war-only.

jzuken commented on db-services/user-service/src/main/java/com/qreal/robots/client/diagrams/DiagramServiceImpl.java:
```diff
@@ -0,0 +1,70 @@
+import org.springframework.stereotype.Service;
+
+/**
+ * Thrift client side of DiagramDbService.
```
it would be nice to make single-line comments occupy only one line of code

jzuken replied:
here, there and everywhere (c)

jzuken commented on db-services/user-service/src/main/java/com/qreal/robots/client/diagrams/DiagramService.java:
```diff
@@ -0,0 +1,15 @@
+
+    /**
+     * Creates root folder for user with specified username.
+     *
```
I doubt that this empty line is in any case useful

jzuken commented on db-services/user-service/src/main/java/com/qreal/robots/client/diagrams/DiagramServiceImpl.java:
```diff
@@ -0,0 +1,70 @@
+        logger.trace("createRootFolder method called with parameters: username = {}", userName);
+        Folder rootFolder = new Folder("root", userName);
+        try {
+            transport.open();
```
createRootFolder() and createFolder() obviously share the same code base

jzuken commented on db-services/user-service/src/main/resources/log4j2.xml:
```diff
@@ -3,7 +3,7 @@
     <Appenders>
         <Appender type="Console" name="Main">
-            <Layout type="PatternLayout" pattern="%d{HH:mm:ss} %C.%M -- %m%n%ex%n"/>
+            <Layout type="PatternLayout" pattern="[user-service] %d{HH:mm:ss} %C.%M -- %m%n%ex%n"/>
```
why "user-service" here? fully qualified class name is not enough?

TanVD replied:
Travis puts all logs in one text. We need either change hierarchy of packages to represent that class from user-service, or add name of service.

jzuken replied:
I think we should reflect the affiliation to the particular service in package hierarchy. Like `com.qreal.robots.userservice.server`.

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuth2UserDetailsLoaderImpl.java:
```diff
@@ -0,0 +1,70 @@
+import java.util.stream.Collectors;
+
+/**
+ * OAuth requests handler.
```
typically it's written here what the class does and who uses it

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuth2UserDetailsLoaderImpl.java:
```diff
@@ -0,0 +1,70 @@
+        if (!userService.isUserExist((String) id)) {
+            return null;
+        } else {
+            com.qreal.robots.database.users.model.User user = userService.findByUserName((String) id);
```
why not just import the class?

TanVD replied:
We already use here User from springframework. I haven't heard about qualified import in java.

jzuken replied:
may be spring's User would have shorter name? :)

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuth2UserDetailsLoaderImpl.java:
```diff
@@ -0,0 +1,70 @@
+
+    @Override
+    public UserDetails getUserByUserId(Object id) {
+        if (!userService.isUserExist((String) id)) {
```
it would be nice to check is the `id` is really a string object

TanVD replied:
We already use here User from springframework. I haven't heard about qualified import in java.

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuth2UserDetailsLoaderImpl.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    @Override
+    public UserDetails updateUser(UserDetails userDetails, Map userInfo) {
```
unchecked operations should be avoided. use `Map<String, Object>` instead

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuth2UserDetailsLoaderImpl.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    @Override
+    public UserDetails createUser(Object id, Map userInfo) {
```
and everywhere else too

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuth2UserDetailsLoaderImpl.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    private List<GrantedAuthority> buildUserAuthority(Set<UserRole> userRoles) {
+        Set<GrantedAuthority> setAuths = userRoles.stream().map(userRole -> new SimpleGrantedAuthority(userRole.
```
ending the line with a dot doesn't look nice here. better start the new line beginning with `new`

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuthBeans.java:
```diff
@@ -0,0 +1,65 @@
+import java.net.URISyntaxException;
+
+@Configuration
+public class OAuthBeans {
```
javadoc is missing

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuthBeans.java:
```diff
@@ -0,0 +1,65 @@
+
+@Configuration
+public class OAuthBeans {
+    private String accessTokenUri = "http://localhost:8080/oauth/token";
```
shouldn't we better store all these port numbers in a property file?

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuthBeans.java:
```diff
@@ -0,0 +1,65 @@
+public class OAuthBeans {
+    private String accessTokenUri = "http://localhost:8080/oauth/token";
+
+    private String userAuthorizationUri = "http://localhost:8080/oauth/authorize";
```
aren't these string final?

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuthBeans.java:
```diff
@@ -0,0 +1,65 @@
+    @Bean
+    @Autowired
+    public OAuth2UserDetailsService<User> oAuth2UserDetailsService(OAuth2ServiceProperties serviceProperties,
+                                                                   OAuth2UserDetailsLoader detailsLoader) throws
```
move `throws` to the next line

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuthBeans.java:
```diff
@@ -0,0 +1,65 @@
+    @Bean(name = "provider")
+    @Autowired
+    public OAuth2AuthenticationProvider oAuth2AuthenticationProvider(OAuth2ServiceProperties serviceProperties,
+                                                                     OAuth2UserDetailsService<User> details) throws
```
and here too

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuthBeans.java:
```diff
@@ -0,0 +1,65 @@
+    @Autowired
+    public OAuth2UserDetailsService<User> oAuth2UserDetailsService(OAuth2ServiceProperties serviceProperties,
+                                                                   OAuth2UserDetailsLoader detailsLoader) throws
+            URISyntaxException {
```
I guess the indentation is screwed here

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/config/SecurityConfig.java:
```diff
@@ -0,0 +1,76 @@
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
```
missing javadoc

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/config/WebConfig.java:
```diff
@@ -0,0 +1,41 @@
+@EnableWebMvc
+@Import({SecurityConfig.class})
+@ComponentScan("com.qreal.robots")
+public class WebConfig extends WebMvcConfigurerAdapter {
```
javadoc

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/config/core/AppInit.java:
```diff
@@ -0,0 +1,44 @@
+import javax.servlet.ServletException;
+import javax.servlet.ServletRegistration;
+
+public class AppInit implements WebApplicationInitializer {
```
javadoc

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/config/core/AppInit.java:
```diff
@@ -0,0 +1,44 @@
+        AnnotationConfigWebApplicationContext dispatcherContext = new AnnotationConfigWebApplicationContext();
+
+        dispatcherContext.scan("com.qreal.robots");
+
```
unnecessary empty line

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/config/core/AppInit.java:
```diff
@@ -0,0 +1,44 @@
+        dispatcherContext.scan("com.qreal.robots");
+
+        dispatcherContext.register(AppInit.class);
+
```
and here too

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/exception/ExceptionHandlerController.java:
```diff
@@ -0,0 +1,28 @@
+import org.springframework.web.servlet.NoHandlerFoundException;
+
+@ControllerAdvice
+public class ExceptionHandlerController {
```
javadoc

jzuken commented on editor-service/src/main/java/com/qreal/robots/common/auth/OAuth2UserDetailsLoaderImpl.java:
```diff
@@ -0,0 +1,70 @@
+ * OAuth requests handler.
+ */
+@Component
+public class OAuth2UserDetailsLoaderImpl implements OAuth2UserDetailsLoader {
```
so these auth classes are basically copied for all other services. can't we reuse them?

jzuken commented on editor-service/src/main/java/com/qreal/robots/database/diagrams/model/DefaultDiagramNode.java:
```diff
@@ -0,0 +1,113 @@
+/**
+ * Diagram's node.
+ */
+public class DefaultDiagramNode implements Serializable {
```
these classes are the same as in diagram DB service, but without annotatons. is there really no way to reuse the code?

jzuken commented on editor-service/src/main/java/com/qreal/robots/database/diagrams/model/LinkProperty.java:
```diff
@@ -0,0 +1,97 @@
+/**
+ * Property of a link.
+ */
+public class LinkProperty implements Serializable {
```
is there a single reason why we need two separate but identical classes for node property and link property?

jzuken commented on editor-service/src/main/java/com/qreal/robots/database/robots/model/Robot.java:
```diff
@@ -0,0 +1,153 @@
+            return false;
+        }
+
+        Robot robot = (Robot) object;
```
these two lines could be merged

jzuken commented on editor-service/src/main/java/com/qreal/robots/database/users/client/UserDetailsServiceImpl.java:
```diff
@@ -0,0 +1,55 @@
+            throws UsernameNotFoundException {
+
+        com.qreal.robots.database.users.model.User user = userService.findByUserName(username);
+
```
too many empty lines here

jzuken commented on editor-service/src/main/java/com/qreal/robots/database/users/client/UserServiceImpl.java:
```diff
@@ -0,0 +1,107 @@
+    public void save(User user) {
+        logger.trace("save method called with parameters: user = {}", user.getUsername());
+        try {
+            transport.open();
```
we could extract a function here that opens the transport, performs an action provided as an argument and closes the transport

TanVD committed:
Source files deleted from 3dparty dir. Jars put instead.

TanVD committed:
Gitignore changed to load jars from 3dparty dir.

TanVD committed:
1.Fixed travis installation of 3dparty libs
2.New hierarchy of packages

jzuken commented on db-services/user-service/src/main/java/com/qreal/wmp/db/user/client/diagrams/DiagramServiceImpl.java:
```diff
@@ -0,0 +1,70 @@
+            logger.trace("createRootFolder method created rootFolder for {}", userName);
+        } catch (TException e) {
+            logger.error("Client DiagramService encountered problem while sending createFolder request with " +
+                    "parameters: newFolder = {}", rootFolder.getFolderName(), e);
```
shouldn't there be two '{}' placeholders?

TanVD replied:
As far as i understand there is no need for another placeholder for Throwable. It will be printed automatically. Here (http://www.slf4j.org/apidocs/index.html ) in docs individual method with such parameters `void error(String msg, Throwable t)`

TanVD committed:
Docs for all services fixed.
OAuth properties files added for editor and dashboard.
A lot of minor improvements.

TanVD committed:
1.Tomcat7 dependency deleted from pom.
2.Few fixes for codacy.

TanVD committed:
Travis fixing. Now using run-war instead of run-war-only.

TanVD committed:
Found bug: Too slow creation of secure random in tomcat7. Trying workaround.

TanVD committed:
Reduced entropy for hash encoder.

TanVD committed:
Problem with random solved.

jzuken commented on .travis.yml:
```diff
@@ -1,4 +1,4 @@
-language: java
+ызкщlanguage: java
```
абырвалг!

TanVD replied:
Упс. Кстати, всё равно ведь сожрал и всё прошёл :)

TanVD committed:
1.Sleep deleted.
2.Now run-war-only used.
3.Minor improvements.

jzuken commented on auth-service/src/main/java/com/qreal/wmp/auth/AppInit.java:
```diff
@@ -0,0 +1,91 @@
+                return;
+            }
+
+            Set<GrantedAuthority> userAuthorities = new HashSet<GrantedAuthority>();
```
I would have moved users creation into separate methods.

jzuken replied:
this method is way too long anyway

jzuken commented on auth-service/src/main/java/com/qreal/wmp/auth/AppInit.java:
```diff
@@ -0,0 +1,91 @@
+            Set<String> grantTypes = new HashSet<String>();
+            grantTypes.add("authorization_code");
+
+            Client dashboardService = new Client("dashboardService", true, "secret", true, scopes, grantTypes,
```
this `new` operator definitely deserves its own method

jzuken commented on auth-service/src/main/java/com/qreal/wmp/auth/config/oauth2/OAuth2ServerConfig.java:
```diff
@@ -0,0 +1,124 @@
+    protected static class ResourceServerConfiguration extends ResourceServerConfigurerAdapter {
+        @Override
+        public void configure(HttpSecurity http) throws Exception {
+            http.
```
this `http.` looks so lonely...

TanVD replied:
My IDEA formats it exactly that way. And i don't really understand why..

jzuken replied:
![](http://i2.kym-cdn.com/entries/icons/original/000/003/617/okayguy.jpg)

jzuken commented on auth-service/src/main/java/com/qreal/wmp/auth/config/oauth2/OAuth2ServerConfig.java:
```diff
@@ -0,0 +1,124 @@
+    }
+
+    @Configuration
+    protected static class Stuff {
```
nice name!

jzuken commented on auth-service/src/main/webapp/WEB-INF/pages/OAuth/error.jsp:
```diff
@@ -0,0 +1,31 @@
+
+<div class="container">
+
+  <h1>OAuth Server OAuth2 Error</h1>
```
this error page is not depressive enough :)

jzuken commented on auth-service/src/main/webapp/WEB-INF/pages/ROLE_ADMIN/usersPanel.jsp:
```diff
@@ -0,0 +1,60 @@
+                <th>Login</th>
+                <th>Password</th>
+            </tr>
+            <c:forEach var="o" items="${users}" varStatus="status">
```
somewhere one martin fowler silently cries sitting in the corner

jzuken committed:
minor formatting fixes

jzuken committed:
Merge branch 'split-robotseditor' of https://github.com/qreal/wmp into split-robotseditor

TanVD committed:
Minor improvements of style.

TanVD committed:
Codacy fixes.

jzuken committed:
renamed db services

jzuken committed:
minor formatting fixes

jzuken committed:
Merge branch 'master' into split-robotseditor
